### ISSUE TITLE
Accuracy differencies on DL4J v0.9.1 and on DL4J v1.0.0-alpha (native CPU, windows)

### ISSUE THREAD

**Issue by ospavel** [deeplearning4j/deeplearning4j#4924]
I have big differences on training the same LSTM model with same training data on DL4J v0.9.1 and on DL4J v1.0.0-alpha (native CPU, windows). 

Code:
```
NeuralNetConfiguration.ListBuilder listBuilder = new NeuralNetConfiguration.Builder()
             .seed((int) (System.currentTimeMillis() / 1000))
             .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
             .updater(new Adam.Builder().learningRate(0.01).build())
             .l2(1e-5)
	     .regularization(true) // remove at 1.0.0-alpha
             .activation(Activation.TANH)
             .weightInit(WeightInit.XAVIER)
             .trainingWorkspaceMode(WorkspaceMode.SEPARATE)
             .inferenceWorkspaceMode(WorkspaceMode.SINGLE)
             .list();
     
int l = 0;
// input
listBuilder.layer(l++, new GravesLSTM.Builder().nIn(43).nOut(8).build());
// hidden
listBuilder.layer(l++, new GravesLSTM.Builder().nIn(8).nOut(8).build());
// output
listBuilder.layer(l, new RnnOutputLayer.Builder(LossFunctions.LossFunction.MCXENT)
                         .activation(Activation.SOFTMAX).nIn(8).nOut(2).build());

MultiLayerNetwork net = new MultiLayerNetwork(listBuilder.pretrain(false).backprop(true).build());
net.init();

ListDataSetIterator trainIterator = new ListDataSetIterator(dataset.asList(), 128);
Evaluation evaluation = new Evaluation(2);
for (int i = 0; i < epochs; i++) {
  trainIterator.reset();
  net.fit(trainIterator);
  if (i % 2 == 0) {
      double score = net.score();
      INDArray predicted = net.output(dataTest.getFeatures(), false, dataTest.getFeaturesMaskArray(), dataTest.getLabelsMaskArray());
      evaluation.reset();
      evaluation.evalTimeSeries(dataTest.getLabels(), predicted, dataTest.getLabelsMaskArray());
	  ...
  }
}  
```

On training with DL4J 0.9.1 I get accuracy 0.68 after 3rd epoch already (0.69 after 30 epoch, score 0.518), but with DL4J 1.0.0-alpha I get 0.62 after 3rd and about 0.67 after 30 epoch (score 0.495). Train data: 137 batches with size 128. The accuracy raises much slower. During several tests.

Why? Maybe during migration need to do some additional changes in model or training process?

**Comment by AlexDBlack** [comment]
There shouldn't be any difference between the releases for RNNs. (All functionality was thoroughly tested, gradient checked etc before both releases, examples were run and performed equivalently for both etc). I'm not seeing anything in the config that would suggest a likely cause here.

I'll have to run both side by side to see. However, given that I know our examples are fine on both versions (I've checked before the 1.0.0-alpha release) I'll also need your dataset to reproduce this. Is it possible to share it?

**Comment by ospavel** [comment]
I saved dataset into file. How I can send it to you?

**Comment by AlexDBlack** [comment]
Feel free to direct message me on gitter: http://gitter.im/AlexDBlack
if it's small you can share it directly via gitter, or you can send me a dropbox/google drive etc link.

**Comment by AlexDBlack** [comment]
Apologies for not getting to this sooner.

After running your code on your data, it appears you were running into a bug on 0.9.1.
Specifically, the default learning rate (0.1) was being used, when .learningRate(x) was not being called.

Consequently, a LR of 0.1 was being used instead of what you set.

Now, if you set .learningRate(0.01) (on the NeuralNetConfiguration builder, not on the Adam builder) on 0.9.1, both versions give the same scores and evaluation.
Similarly, setting LR of 0.1 in 1.0.0-alpha causes both versions to give the same scores and evaluation.
